tests/core/test_accounts.py

```diff
@@ -322,6 +322,63 @@ def test_eth_account_sign(acct, message, key, expected_bytes, expected_hash, v,
     assert account.signHash(msghash) == signed
 
 
+@pytest.mark.parametrize(
+    'message, message_type',
+    (
+        ("hello world", "string"),
+        (b'hello world', "bytes"),
+        ("68656c6c6f20776f726c64", "hexstr"),
+    )
+)
+def test_eth_account_sign_data_with_intended_validator(message, message_type, acct):
+    account = acct.create()
+
+    if message_type == "string":
+        primitive = None
```

Comment:
style nit: You could assign these to a variable like: `kwargs = {'text': message}`, and then pass `**kwargs` to `defunct_hash_message` below. Then, you'd avoid having to assign `None` to the args you weren't using.